test/common/quic/active_quic_listener_test.cc

```diff
@@ -608,9 +608,15 @@ TEST_P(ActiveQuicListenerTest, EcnReportingIsEnabled) {
   Api::SysCallIntResult rv;
   if (*version == Network::Address::IpVersion::v6) {
     rv = socket.getSocketOption(IPPROTO_IPV6, IPV6_RECVTCLASS, &optval, &optlen);
-  } else {
-    rv = socket.getSocketOption(IPPROTO_IP, IP_RECVTOS, &optval, &optlen);
+    EXPECT_EQ(rv.return_value_, 0);
+    EXPECT_EQ(optval, 1);
+    rv = socket.getSocketOption(IPPROTO_IPV6, IPV6_V6ONLY, &optval, &optlen);
+    EXPECT_EQ(rv.return_value_, 0);
+    if (optval) {
+      return;
+    }
   }
+  rv = socket.getSocketOption(IPPROTO_IP, IP_RECVTOS, &optval, &optlen);
```

Comment:
Could you comment in what situation we should check TOS via IP_RECVTOS option?

Reply:
We're not checking TOS, we're checking if RECVTOS was set properly.

Reply:
Anyway, please comment what situation (seems like v4 or dual stack v6) we should check IP_RECVTOS instead of the other one.

Reply:
For dual stack socket, do both IPV6_RECVTCLASS and IP_RECVTOS populates optval with a non-zero value? Does it still make sense to check IPV6_RECVTCLASS in that case?

Reply:
Yes, although the behavior is a little different on Apple platforms. This test will make sure that there's no problem there.

Reply:
For such special case, I suggest add more comment about the platform behavior difference.

Reply:
This need more comments.

Reply:
I've updated the comment. These platform APIs are very poorly documented and I'm not fully sure what to expect. What I know about them comes from trial and error in Chromium. This test is more about making sure that if there is a platform problem we detect it.

To be honest, I'm surprised that all the Apple tests passed.